### PR TITLE
[ci] Check whitespace on PRs

### PR DESCRIPTION
**Pull request by ricab**

### CODE REVIEW

**Comment by Sploder12** [comment]
clang-format provides [InsertNewlineAtEOF](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#insertnewlineateof) which could be useful, but it's from version 16 so we'd have to update our versions.

**Review by andrei-toterman — COMMENTED** [review]
I see this runs only on the Linux workflow. When this runs on the private repo, will it work?

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -28,6 +28,12 @@ jobs:
...
         fetch-depth: 2
 
+    - name: Check whitespace
+      uses: ./.github/actions/whitespace
+
+    - name: Check final newline
+      uses: qwrtln/check-final-newline@v1.0.0
+
     - name: Lint the code
       uses: ./.github/actions/lint
```
Comment (andrei-toterman) [inline .github/workflows/linux.yml:38]: Instead of adding these as new actions, can't they be integrated into the `./github/actions/lint`?

Comment (ricab) [reply .github/workflows/linux.yml]: Yep, good idea. Done.

**Review by andrei-toterman — APPROVED** [review]
So the public side is all good and I am ready to approve. However, I see that on the private side the lint fails on the `packaging/windows/template.rtf` file because it does not have a new line at the end of the file. So I will approve this PR, but we need a private PR which fixes that thing as well.

**Pull request merged**